PolyEngine/UnitTests/CMakeLists.txt

```diff
@@ -0,0 +1,85 @@
+set(POLYTESTS_SRCS
+	Src/AllocatorTests.cpp
+	Src/AngleTests.cpp
+	Src/BasicMathTests.cpp
+	Src/DynarrayTests.cpp
+	Src/EnumUtilsTests.cpp
+	Src/main.cpp
+	Src/MatrixTests.cpp
+	Src/QuaternionTests.cpp
+	Src/QueueTests.cpp
+	Src/VectorTests.cpp
+)
+
+if(NOT TARGET Catch) #TODO(vuko): what about our embedded copy of Catch? How to integrate sanely? And should we?
+	include(ExternalProject)
+	if(IS_TRAVIS_BUILD)
+		set(FETCH_EXTERNAL_CATCH GIT_REPOSITORY https://github.com/philsquared/Catch.git GIT_TAG 0c015aa8878db0b582afa95bb9ef8ebf97baf232)
+	elseif(WIN32)
+		set(FETCH_EXTERNAL_CATCH URL https://github.com/philsquared/Catch/archive/v1.8.2.zip URL_HASH MD5=d929768234f22382a32d9bfd76b32355)
+	else()
+		set(FETCH_EXTERNAL_CATCH URL https://github.com/philsquared/Catch/archive/v1.8.2.tar.gz URL_HASH MD5=23562b3d3c8cc0a191ca74580320429a)
+	endif()
+	ExternalProject_Add(
+		Catch-External
+		PREFIX ${CMAKE_BINARY_DIR}/external/Catch
+		${FETCH_EXTERNAL_CATCH}
+		CONFIGURE_COMMAND ""
+		BUILD_COMMAND ""
+		INSTALL_COMMAND ${CMAKE_COMMAND} -E copy ${CMAKE_BINARY_DIR}/external/Catch/src/Catch-External/single_include/catch.hpp
+													${CMAKE_BINARY_DIR}/external/Catch/include/catch.hpp
+	)
+	add_library(Catch INTERFACE)
+	add_dependencies(Catch Catch-External)
+	target_include_directories(Catch INTERFACE ${CMAKE_BINARY_DIR}/external/Catch/include)
+endif(NOT TARGET Catch)
+
+add_executable(polytests ${POLYTESTS_SRCS})
+target_link_libraries(polytests polycore polyengine Catch)
+#TODO(vuko): you have to manually add a case here every time you add a new test to Catch... Maybe grep the files looking for TEST_CASE then generate?
+add_test(NAME "Pool-allocator"                                COMMAND polytests "Pool allocator")
```

Comment:
No this just adds our tests to CTest, so one can run `make tests` and get a sane output. You can still manually run the resulting executable though, so it's mostly a convenience thing.